### DISCUSSION TITLE
How to remove site replication bucket bandwidth limit?

### DISCUSSION THREAD

**Discussion by schnapsidee** [minio/minio#19094] [Q&A] [answered]
Hi, sorry, I wasn't sure if the question belonged into the mc repo or here.

I used `mc admin replicate update --bucket-bandwidth` to limit the bandwidth for site replication. 
I now no longer need this limit, but the [documentation](https://min.io/docs/minio/linux/reference/minio-mc-admin/mc-admin-replicate.html#mc.admin.replicate.update) doesn't mention a way to completely remove the limit. Is this just missing, or is there no way to remove it without removing the entire site replication (which would require a resync of all data)? 

I suppose I could just set a very high limit, but I would prefer it if I could just remove the unnecessary config.

**Comment by schnapsidee** [comment]
It looks like you can reset the default bucket bandwidth by using `mc admin replicate update --bucket-bandwidth 0`
This only removes the bandwidth limit for new buckets, though. In fact, the bandwidth limit — once set — can't seem to be changed at all.
I'm not entirely sure why it was designed that way, but it looks like the first limit that's set, is the one you're stuck with and the only way to change it is to remove the replication rule and create a new one (which you can't do in a site-replication setup). 

[This](https://github.com/minio/minio/blob/master/cmd/site-replication.go#L1130) seems to be the relevant piece of code. The bandwidth limit only gets updated if none was set before.

```go
updateBW = peer.DefaultBandwidth.Limit != 0 && target.BandwidthLimit == 0
if (target.URL().String() != peer.Endpoint) || updateBW {
    updateTgt = true
    targetToUpdate = target
}

[...]

if updateBW {
    targetToUpdate.BandwidthLimit = int64(peer.DefaultBandwidth.Limit)
}
```

**Reply by harshavardhana** [reply]
@poornas ^^

**Reply by poornas** [reply]
per bucket bandwidth setting is not overridden by the site level default - the site level setting is more of a template to apply to new buckets or previously created buckets with no bandwidth limits. This  allows prioritizing available replication bandwidth at will to certain buckets

To answer your original question,  there is no option to unset bandwidth limit. You can set a very high limit to achieve this.

**Reply by schnapsidee** [reply]
Thank you, @poornas. I understand the reasoning behind why you do it this way now, but I don't think there's a way to actually change the bandwidth limit on existing rules in a site replication setup, then. 

At least, I couldn't find a way to do it. There's `mc replicate update` but when I try to use this to set a new limit for an existing rule I get `mc: <ERROR> no valid credentials were detected. Invalid arguments provided` . 

Am I doing something wrong? I set an mc alias with the root user and used replication id and bucket destination from `mc replicate ls alias/bucket`

So this is what my command looked like: `mc replicate update test/test --id "site-repl-33a9197f-104f-492d-9724-415dc33b1b37" --remote-bucket "arn:minio:replication::33a9197f-104f-492d-9724-415dc33b1b37:test" --bandwidth 1G`

**Reply by poornas** [reply, marked as answer]
you will need to pass the --remote-bucket differently. Should be in http(s)://<ak>:<sk>@endpoint/<target-bucket>](https://foobar:newpassword@minio.siteb.example.com/targetbucket format

**Reply by schnapsidee** [reply]
That worked. Thank you.
